### PR TITLE
dense.jl autotangent

### PR DESCRIPTION
**Pull request by oxinabox**
part of #360

### CODE REVIEW

File: test/rulesets/LinearAlgebra/dense.jl [hunk 1]
```diff
@@ -66,25 +37,31 @@
...
         end
+
         @testset "$F{Vector{$T}}" for T in (Float64, ComplexF64), F in (Transpose, Adjoint)
-            n = 3
-            x, ẋ, x̄ = F(randn(T, n)), F(randn(T, n)), F(randn(T, n))
+            test_frule(pinv, F(randn(T, 3)) ⊢ F(randn(T, 3)))
+            test_rrule(pinv, F(randn(T, 3)))
+
+            # Check types.
+            # TODO: Do we need this still?
```
Comment (oxinabox) [inline test/rulesets/LinearAlgebra/dense.jl:56]: @willtebbutt do you know?

Comment (willtebbutt) [reply test/rulesets/LinearAlgebra/dense.jl:56]: I have absolutely no idea 😬

**Comment by oxinabox** [comment]
Shall we merge this?

**Review by willtebbutt — APPROVED** [review]
LGTM

**Pull request merged**